### PR TITLE
Properly escape user input by using Open3 capture methods

### PR DESCRIPTION
**Pull request by ericfranz**
Also remove 'active_users' from developer interface which was broken as of Passenger 6 (and not really used anyways).

### CODE REVIEW

File: apps/myjobs/test/models/configuration_singleton_test.rb [hunk 1]
```diff
@@ -65,6 +69,10 @@ def config_via_runner(env: 'development', envvars: '')
   end
 
   test "loading custom AweSim external config in production env" do
+    # TypeError: incompatible marshal file format (can't be read)
+    #     format version 4.8 required; 32.32 given
+    skip
```
Comment (johrstrom) [inline apps/myjobs/test/models/configuration_singleton_test.rb:74]: What's this? I found this [rubygems ticket on for the same](https://github.com/rubygems/rubygems/issues/1589). Could it be an environment issue you have?  I'll pull to replicate too.

**Comment by ericfranz** [comment]
Looks like Open3.capture methods do not like spaces in the arguments.

These all fail

```
# these fail with unknown option, rsync error: syntax or usage error
o,e,s = Open3.capture3 'rsync', "-r --exclude='.svn' --exclude='.git' --exclude='.gitignore' --filter=':- .gitignore'", 'foo/','bar'
o,e,s = Open3.capture3 'rsync', "-r --exclude='.svn' --exclude='.git' --exclude='.gitignore'", 'foo/','bar'

# these fail with unknown filter rule
o,e,s = Open3.capture3 'rsync', '-r', "--exclude='.svn'", "--exclude='.git'", "--exclude='.gitignore'", "--filter=':- .gitignore'", 'foo/','bar'
o,e,s = Open3.capture3 'rsync', '-r', "--exclude='.svn'", "--exclude='.git'", "--exclude='.gitignore'", "--filter=':-", ".gitignore'", 'foo/','bar'

# this succeeds
o,e,s = Open3.capture3 'rsync', '-r', "--exclude='.svn'", "--exclude='.git'", "--exclude='.gitignore'", 'foo/','bar'
```

File: apps/myjobs/test/models/configuration_singleton_test.rb (hunk 1 above)
Comment (johrstrom) [reply apps/myjobs/test/models/configuration_singleton_test.rb:74]: Scratch that, it likely doesn't like the env file and has nothing to do with rubygems.

I can't replicate on Linux, either on my machine or in pitzer. In any case, I think we should keep it just to safeguard us from future mistakes.

File: apps/myjobs/test/models/filesystem_test.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+require 'test_helper'
+
+class FilesystemTest < ActiveSupport::TestCase
+  test "copy_dir doesn't allow command injection" do
+    o,e,s = Filesystem.new.copy_dir("/dev/null';echo INJECTED;'", '/dev/null')
+    refute_match /INJECTED/, o
```
Comment (johrstrom) [inline apps/myjobs/test/models/filesystem_test.rb:6]: I get the same error  that travis does. I guess we need to rescue or expect the exception here?

File: apps/myjobs/test/models/configuration_singleton_test.rb (hunk 1 above)
Comment (ericfranz) [reply apps/myjobs/test/models/configuration_singleton_test.rb:74]: So you are saying if you revert https://github.com/OSC/ondemand/pull/702/commits/8d018238d699486db25a09b8593dca0be48a9646 you have no problem?

Comment (johrstrom) [reply apps/myjobs/test/models/configuration_singleton_test.rb:74]: Yep. I reverted on Pitzer and my Linux machine and couldn't get the same error.  Plus Travis in `master` would see it too right?

Comment (ericfranz) [reply apps/myjobs/test/models/configuration_singleton_test.rb]: reverted with c7182fba9fbc4a8faeaef9d1fbcd6e8a7b127d67

**Pull request merged**